LibZipSharp/Xamarin.Tools.Zip/ZipArchive.cs

```diff
@@ -803,17 +828,31 @@ internal static unsafe Int64 stream_callback (IntPtr state, IntPtr data, UInt64
 					}
 
 				case SourceCommand.SeekWrite:
-					Native.zip_error_t error;
-					Int64 offset = Native.zip_source_seek_compute_offset ((UInt64)destination.Position, (UInt64)destination.Length, data, len, out error);
-					if (offset < 0) {
-						return offset;
+					Native.zip_source_args_seek_t args;
+					if (!Native.ZipSourceGetArgs (data, len, out args)) {
+						return -1;
+					}
+
+					long firstOffset, secondOffset;
+					if (args.offset > Int64.MaxValue) {
+						// Stream.Seek uses a signed 64-bit value for the offset, we need to split it up
```

Comment:
I'm a bit confused on this point?

Reply:
args.offset is an `ulong`, but `Seek` takes a long, and if you pass it a negative value it will search back. So if the passed value exceeds the range of `long`, we need to split it into two seeks.

Reply:
Never mind I think I get it now. The offset is longer than the signed value (long/Int64) which the Stream.Seek API uses.